server/app/views/admin/programs/ProgramBlockReadOnlyView.java

```diff
@@ -0,0 +1,312 @@
+            });
+    return new ArrayList<DomContent>(
+        Arrays.asList(blockInfoDisplay, predicateDisplay, programQuestions));
+  }
+
+  protected DivTag renderPredicate(
+      ProgramDefinition programDefinition /* needed for subclasses */,
+      BlockDefinition blockDefinition,
+      ImmutableList<QuestionDefinition> questions) {
+
+    Optional<PredicateDefinition> predicate = blockDefinition.visibilityPredicate();
+    String currentBlockStatus =
+        predicate.isEmpty()
+            ? "This screen is always shown."
+            : predicate.get().toDisplayString(blockDefinition.name(), questions);
+
+    return div()
+        .withClasses("my-4")
+        .with(div("Visibility condition").withClasses("text-lg", "font-bold", "py-2"))
+        .with(div(currentBlockStatus).withClasses("text-lg", "max-w-prose"));
+  }
+
+  /** Renders an individual question to be shown in the Block panel. */
+  protected DivTag renderQuestion(
+      ProgramDefinition programDefinition /* needed for subclasses */,
+      BlockDefinition blockDefinition,
+      int questionIndex) {
+    ImmutableList<ProgramQuestionDefinition> blockQuestions =
+        blockDefinition.programQuestionDefinitions();
+    ProgramQuestionDefinition question = blockQuestions.get(questionIndex);
+
+    QuestionDefinition questionDefinition = question.getQuestionDefinition();
+    DivTag ret =
+        div()
+            .withClasses(
+                ReferenceClasses.PROGRAM_QUESTION,
+                "my-2",
+                "border",
+                "border-gray-200",
+                "px-4",
+                "py-2",
+                "flex",
+                "gap-4",
+                "items-center",
+                StyleUtils.hover("text-gray-800", "bg-gray-100"));
+
+    SvgTag icon =
+        Icons.questionTypeSvg(questionDefinition.getQuestionType())
+            .withClasses("shrink-0", "h-12", "w-6");
+    String questionHelpText =
+        questionDefinition.getQuestionHelpText().isEmpty()
+            ? ""
+            : questionDefinition.getQuestionHelpText().getDefault();
+    DivTag content =
+        div()
+            .withClass("flex-grow")
+            .with(
+                p(questionDefinition.getQuestionText().getDefault()),
+                p(questionHelpText).withClasses("mt-1", "text-sm"),
+                p(String.format("Admin ID: %s", questionDefinition.getName()))
+                    .withClasses("mt-1", "text-sm"));
+    return ret.with(icon, content);
+  }
+
+  @Override
+  protected String getEditButtonText() {
+    return "Edit program";
+  }
+
+  // TODO when the ProgramBlockReadOnlyView is used, clicking the button should lead to the
+  // ProgramBlockEditView.
+  @Override
+  protected String getButtonUrl(ProgramDefinition programDefinition) {
+    return null;
+  }
+
+  @Override
+  protected BadgeStatus getBadgeStatus() {
+    return BadgeStatus.ACTIVE;
```

Comment:
That was my understanding from my conversation with Cassandra.
A Draft would open the edit view
Unless I'm mistaken, an obsolute or deleted version isn't usually visible

By "enforced" do you mean something other than implementing the behaviour defines which click opens what version?

Reply:
"Unless I'm mistaken, an obsolute or deleted version isn't usually visible"

Currently only a draft is visible so nothing else is usually visible, It's conceivable we may want to allow viewing an obsolete Program as there will be applications on them and an Admin could want to view it, but seemingly it's not something that's being asked for.

"by enforced" I didn't realize this was in effect "ActiveProgramBlock....." I thought it might be more general, eg ReadOnly of the draft to prevent accidents etc etc.  Don't think that's a blocker though, but maybe add a comment here that this is intended to only be used for Active programs.

Reply:
I'm going for option 1 of your suggestions with the following reasoning:
From a conversation with Cassandra I know that the ability viewing obsolete programs was discarded at some point because it wasn't used or requested enough. It therefore seems unlikely that there will be requests for it soon. A 1:1 mapping of view to state therefore seems the simplest implementation that meets the medium term needs.

I've renamed the class. 
When I implement the navigation, the checking of the state will be implemented by following the same pattern as for the edit view (and in line with what you suggest in your comment). In other words, inside the AdminProgramBlocksController there will be a method view(), similar to create() and edit(), which also uses the RequestChecker to validate state.

I'll do some manual testing to see how exactly this error is surfaced to the user for the edit case before implementing it for the view case in the same way.